### DISCUSSION TITLE
how to request spending transaction of a specific txid:vout with json rpc?

### DISCUSSION THREAD

**Discussion by antonilol** [romanz/electrs#777] [Q&A]
How can i get the transaction id and input index of the input that spends a specific transaction output, using the json rpc?

**Comment by romanz** [comment]
Currently, Electrum protocol doesn't implement this query (https://electrumx-spesmilo.readthedocs.io/en/latest/protocol-methods.html).
But the existing index contains the relevant data to answer such query - see https://github.com/romanz/electrs/blob/master/doc/schema.md#transaction-inputs-index-spending.

**Reply by antonilol** [reply]
do you mean the data is indexed, but there is no way to get it out other that stopping electrs and reading the raw database (i think i cant read the database with electrs running without risking a data race)?

**Reply by romanz** [reply]
> the data is indexed, but there is no way to get it out other that stopping electrs and reading the raw database

yes.

**Reply by romanz** [reply]
OTOH, you can consider using Blockstream's fork of electrs, which does support this RPC:
https://github.com/blockstream/esplora/blob/master/API.md#get-txtxidoutspendvout

**Reply by antonilol** [reply]
i know it exists, but it is way to heavy for my home node

**Reply by antonilol** [reply]
could you make a query to get this data? you could name it `electrs.blockchain.get_spending`, or something to indicate it is not part of the electrum protocol spec

**Reply by antonilol** [reply]
this would be really useful for home node users running https://github.com/mempool/mempool (i do myself) to click red arrows to go to spending transactions

**Comment by romanz** [comment]
Please try #454.

**Reply by antonilol** [reply]
it only returns the spending transaction's txid, not the index of the input spending it. i asked for this at the electrum protocol 2.0 pr some months ago ([link](https://github.com/spesmilo/electrumx/pull/90#discussion_r1049749939)) and implemented it on my fork to test it with mempool https://github.com/antonilol/electrs/tree/p2p-outpoint-rpc-input-index
